### PR TITLE
Mob 1038 new bottom navigation

### PR DESCRIPTION
**Pull request by ecotopian**
This is ready for review now. It's quite big so take your time. It has quite some changes especially on UI layer side and combines many subtasks into one PR.

### CODE REVIEW

File: Client/Frontend/Home/FirefoxHomeViewController.swift [hunk 1]
```diff
@@ -243,37 +278,46 @@ class FirefoxHomeViewController: UICollectionViewController, HomePanel {
...
 
         func cellHeight(_ traits: UITraitCollection, width: CGFloat) -> CGFloat {
             switch self {
+            case .personalCounter: return UIFontDescriptor.preferredFontDescriptor(withTextStyle: .caption2).pointSize + 28 + 32
             case .promo: return 230
-            case .treeCounter: return 130
+            case .logo: return 100
+            case .search: return UIFontDescriptor.preferredFontDescriptor(withTextStyle: .body).pointSize + 25 + 16
+            case .treeCounter:
+                return 30 + UIFontDescriptor.preferredFontDescriptor(withTextStyle: .subheadline).pointSize + 2 + 16 //TODO: make dynamic
```
Comment (ecotopian) [inline Client/Frontend/Home/FirefoxHomeViewController.swift:316]: remove TODO

File: Client/Frontend/Home/FirefoxHomeViewController.swift [hunk 2]
```diff
@@ -432,6 +467,17 @@ extension FirefoxHomeViewController: UICollectionViewDelegateFlowLayout {
...
 
+    override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        guard let searchbarCell = searchbarCell else { return }
+        delegate?.home(self, didScroll: searchbarCell.frame.minY, offset: scrollView.contentOffset.y)
+    }
+
+    override func scrollViewWillBeginDragging(_ scrollView: UIScrollView) {
+        let velocity = scrollView.panGestureRecognizer.velocity(in: scrollView.superview)
+        delegate?.home(self, willBegin: velocity)
+
```
Comment (ecotopian) [inline Client/Frontend/Home/FirefoxHomeViewController.swift:478]: remove empty line

File: Client/Frontend/Widgets/PhotonActionSheet/AppMenu.swift [hunk 3]
```diff
@@ -7,10 +7,28 @@ import Account
...
+        }
+
+        let myImpact = PhotonActionSheetItem(title: .localized(.myImpact), iconString: "myImpact") { _, _ in
+            (vcDelegate as? BrowserViewController)?.presentEcosiaWorld()
+        }
+
+        return [openHomePage, myImpact]
+    }
+
+
```
Comment (ecotopian) [inline Client/Frontend/Widgets/PhotonActionSheet/AppMenu.swift:28]: remove empty line

File: Client/Ecosia/UI/NTP/SearchbarCell.swift [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+protocol SearchbarCellDelegate: AnyObject {
+    func searchbarCellPressed(_ cell: SearchbarCell)
+}
+
+class SearchbarCell: UICollectionViewCell, Themeable {
```
Comment (vauxhall) [inline Client/Ecosia/UI/NTP/SearchbarCell.swift:11]:
```suggestion
final class SearchbarCell: UICollectionViewCell, Themeable {
```

**Review by vauxhall — APPROVED** [review]
That was a lot, but all of them amazing improvements 💯

**Review by vauxhall — APPROVED** [review]
💯

**Pull request merged**